### PR TITLE
MSPSDS-470: Add custom Keycloak email provider to use GOV.UK Notify service

### PR DESCRIPTION
**Pull request by tom-bell-softwire**
Use the [GOV.UK Notify](https://www.notifications.service.gov.uk) service to send account verification and password reset emails from Keycloak.

For now, system event emails (e.g. for notifying admins about failed logins) are not supported, but these would be simple to add if needed in the future, by creating a new email template and implementing the relevant method in `NotifyEmailTemplateProvider`.

### CODE REVIEW

**Comment by tom-bell-softwire** [comment]
Looks like the Notify service supports integration tests by using ["test" API keys](https://docs.notifications.service.gov.uk/java.html#test), so we should be able to use one of these.

File: keycloak/providers/.gitignore [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Taken from: https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore
```
Comment (broder) [inline keycloak/providers/.gitignore]: We could probably just update the root `.gitignore` to include the Java stuff. We created that using gitignore.io

File: keycloak/providers/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# GOV.UK Notify Email Provider for Keycloak
+
+
+## Getting Setup
```
Comment (broder) [inline keycloak/providers/README.md]: It might be worth adding something brief here.

Comment (tom-bell-softwire) [reply keycloak/providers/README.md]: Good call.

**Pull request merged**